### PR TITLE
docs: demo/dev guide - add running argocd agent components in clusters using open cluster management

### PR DESCRIPTION
**Pull request by mikeshng**
Enhance the demo/dev guide docs with steps to run the agent components using [Open Cluster Management](https://open-cluster-management.io/) and it's [addon framework](https://open-cluster-management.io/docs/concepts/addon/).

### CODE REVIEW

File: hack/demo-env/apps/autonomous-guestbook.yaml [hunk 1]
```diff
@@ -15,5 +13,7 @@ spec:
     server: https://kubernetes.default.svc
     namespace: guestbook
   syncPolicy:
+    automated:
+      prune: true
```
Comment (jannfis) [inline hack/demo-env/apps/autonomous-guestbook.yaml]: For demo purposes, I think this should stay on manual.

File: hack/demo-env/apps/managed-guestbook.yaml [hunk 2]
```diff
@@ -13,5 +13,7 @@ spec:
     server: https://kubernetes.default.svc
     namespace: guestbook
   syncPolicy:
+    automated:
+      prune: true
```
Comment (jannfis) [inline hack/demo-env/apps/managed-guestbook.yaml]: For demo purposes, I think this should stay on manual.

**Review by jannfis — COMMENTED** [review]
Nice! I just have a minor comment, PTAL below

File: hack/demo-env/apps/autonomous-guestbook.yaml (hunk 1 above)
Comment (mikeshng) [reply hack/demo-env/apps/autonomous-guestbook.yaml]: Removed as suggested. Thanks.

File: hack/demo-env/apps/managed-guestbook.yaml (hunk 2 above)
Comment (mikeshng) [reply hack/demo-env/apps/managed-guestbook.yaml]: Removed as suggested. Thanks.

**Review by jannfis — APPROVED** [review]
Freaking awesome :heart: 

LGTM.

**Pull request merged**